pkg/helmreconciler/reconciler.go

```diff
@@ -132,16 +133,21 @@ func (h *HelmReconciler) processRecursive(manifests ChartManifestsMap) *v1alpha2
 				<-s
 				log.Infof("Dependency for %s has completed, proceeding.", c)
 			}
-
+			mu.Lock()
 			if _, ok := out.Status[c]; !ok {
 				out.Status[c] = &v1alpha2.InstallStatus_VersionStatus{}
 			}
 			out.Status[c].Status = v1alpha2.InstallStatus_NONE
+			mu.Unlock()
```

Comment:
Nice catch! You can reduce the amount of locking by keeping the state in a local variable and write it back to the Status map in just one place at the end.

Reply:
Thanks for the suggestion. Fixed!